gdal/ogr/ogrsf_frmts/mssqlspatial/ogr_mssqlspatial.h

```diff
@@ -448,6 +452,8 @@ class OGRMSSQLSpatialDataSource final: public OGRDataSource
     int                *panSRID;
     OGRSpatialReference **papoSRS;
 
+    OGRMSSQLSpatialTableLayer *poLayerInCopyMode;
```

Comment:
Doesn't appear to be initialized by constructor. You can initialize it here with C++11 initialization style:
poLayerInCopyMode = nullptr